runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java

```diff
@@ -28,12 +31,14 @@
 import javax.inject.Inject;
 import javax.inject.Named;
 
-public class CoreValidationsProvider implements ValidationsProvider {
+public class CoreValidationsProvider implements ValidationsProvider, ArtifactAstGraphDependencyProviderAware {
 
   private ClassLoader artifactRegionClassLoader;
 
   private boolean ignoreParamsWithProperties;
 
+  private ArtifactAstDependencyGraphProvider artifactAstDependencyGraphProvider = new CachingArtifactAstDependencyGraphProvider();
```

Comment:
Let's discuss about declaring this as `@Inject private Supplier<ArtifactAstDependencyGraph>`

Reply:
Already discussed added comment and what is needed to not fail before the creation of the validation context.